pkg/volume/emptydir/empty_dir.go

```diff
@@ -310,26 +311,35 @@ func (ed *emptyDir) setupHugepages(dir string) error {
 // getPageSizeMountOptionFromPod retrieves pageSize mount option from Pod's resources
 // and validates pageSize options in all containers of given Pod.
 func getPageSizeMountOptionFromPod(pod *v1.Pod) (string, error) {
-	pageSizeFound := false
+	var (
+		pageSizeFound bool
+		retErr        error
+	)
 	pageSize := resource.Quantity{}
+
 	// In some rare cases init containers can also consume Huge pages.
-	containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
-	for _, container := range containers {
+	podutil.VisitContainers(&pod.Spec, func(c *v1.Container) bool {
 		// We can take request because limit and requests must match.
-		for requestName := range container.Resources.Requests {
+		for requestName := range c.Resources.Requests {
```

Comment:
I think adding ephemeral containers to this list is a noop since according to the kep, ephemeral containers cannot set resources?